lnrpc/rpc.proto

```diff
@@ -2909,12 +2909,39 @@ message ListPaymentsRequest {
     that pending payments, as well as failed payments will show up if this
     field is set to True.
     */
-    bool include_incomplete = 1;
+    bool include_incomplete = 1 [json_name = "include_incomplete"];
+
+    /**
+    The index of a payment that will be used as either the start or end of a
+    query to determine which payments should be returned in the response.
+    */
+    uint64 index_offset = 2 [json_name = "index_offset"];
+
+    /// The max number of payments to return in the response to this query.
+    uint64 num_max_payments = 3 [json_name = "num_max_payments"];
+
+    /**
+    If set, the payments returned will result from seeking backwards from the
+    specified index offset. This can be used to paginate backwards.
+    */
+    bool reversed = 4 [json_name = "reversed"];
 }
 
 message ListPaymentsResponse {
     /// The list of payments
-    repeated Payment payments = 1;
+    repeated Payment payments = 1 [json_name = "payments"];
+
+    /**
+    The index of the first item in the set of returned payments. This can be
+    used to seek backwards, pagination style.
```

Comment:
To seek backwards, do I need to call `ListPayments` again with `index_offset` set to `first_index_offset - 1`?

Reply:
You just need to take the `first_index_offset` to continue pagination.